.github/workflows/instructlab-knowledge-e2e.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow is designed to test the notebooks in "notebooks/instructlab-knowledge"
+# This job can be coppied for end to end tests of notebooks in other working directories
+
+name: Instructlab Knowledge End to End Tests
+
+# trigger on open pull requests only
+on:
+    pull_request:
+        branches:
+          - "main"
+        types: [opened, reopened]
+
+env:
+    # dependencies that need to be installed using pip for testing to work
+    TESTING_PIP_DEPENDENCIES: "papermill nbformat"
+
+jobs:
+    end_to_end:
+        strategy:
+            matrix:
+                # add notebooks within the "notebooks/instructlab-knowledge" repo to end to end test here
+                notebooks_to_test: ["instructlab-knowledge.ipynb"]
+        runs-on: ubuntu-latest
+        env:
+            # customize the workflow here
+            API_KEY: ""
+            API_URL: "http://127.0.0.1:11434/v1"
+            MODEL_ID: "Mixtral-8x7B" # must be open-AI compatible when using inference mock
```

Comment:
Do we need a full Mixtral model unless we are doing quality testing?

Reply:
That doesn't actually do anything, its just so it serves something back. We can make that whatever we want.